KeyValue.cpp

```diff
@@ -658,10 +662,23 @@ void KeyValue::BuildData(char*& destBuffer)
 		}
 		else
 		{
-			memcpy(destBuffer, current->data.leaf.value.string, current->data.leaf.value.length);
-			destBuffer[current->data.leaf.value.length] = '\0';
+			char* string = current->data.leaf.value.string;
+			size_t size = current->data.leaf.value.length;
+			for ( int j = 0; j < size; j++ )
+			{
+				char c = *string++;
+				if ( c == '\\' && ( *string == '\\' || *string == '"' ) )
+				{
+					destBuffer[j] = *string++;
```

Comment:
KeyValues1 supports a number of different escape codes, not just `\\` and `\"`. 
https://github.com/ValveSoftware/source-sdk-2013/blob/0d8dceea4310fde5706b3ce1c70609d72a38efdf/mp/src/tier1/utlbuffer.cpp#L57

This includes
```
\n
\t
\v
\b
\r
\f
\a
\\
\?
\'
\"
```